src/main/scala/chisel3/util/BoringUtils.scala

```diff
@@ -0,0 +1,59 @@
+// See LICENSE for license details.
+
+package chisel3.util
```

Comment:
I suspect this should be marked experimental, what do you think @seldridge?

Tbf we need to do a better job of "promoting" things from experimental which we haven't done yet

Reply:
Experimental seems fine with me. Where should this live, then? Should it be `package chisel3.experimental` or should it live someplace else and use the `experimental` package object?

Reply:
I think maybe a new package `chisel3.util.experimental`? I think we should keep `chisel3.experimental` for things that need to be in core.

Reply:
That seems reasonable to me.